docs/edgeql/select.rst

```diff
@@ -1026,6 +1026,70 @@ exclusively fetch the ``Movie.characters`` of type ``Hero``.
     ...
   }
 
+Accessing types in polymorphic queries
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+While the type of an object is displayed inside polymorphic queries, this
+is simply a convenience of the REPL and not a property that can be accessed.
+This is particularly noticeable if you cast an object to ``json``, making it
+impossible to determine the type if the query is polymorphic:
+
+.. code-block:: edgeql-repl
+
+    db> select Person limit 1;
+    {default::Villain {id: 6c22bdf0-5c03-11ee-99ff-dfaea4d947ce}}
+    db> select <json>Person limit 1;
+    {Json("{\"id\": \"6c22bdf0-5c03-11ee-99ff-dfaea4d947ce\"}")}
+
+The type of an object is found inside ``__type__`` which is a link that
+carries various information about a type, including its ``name``.
+
+.. code-block:: edgeql-repl
+
+    db> select <json>Person { 
+    ...  __type__: {
+    ...    name 
+    ...    } 
+    ...  } limit 1;
+    {Json("{\"__type__\": {\"name\": \"default::Villain\"}}")}
+
+This information can be pulled into the top level by assigning a name to
+the ``name`` property inside ``__type__``:
+
+    db> select <json>Person { n := .__type__.name } limit 1;
+    {Json("{\"n\": \"default::Villain\"}")}
```

Comment:
```suggestion
.. code-block:: edgeql-repl

    db> select <json>Person { type := .__type__.name } limit 1;
    {Json("{\"type\": \"default::Villain\"}")}
```

- Add `code-block` directive so it builds
- Change `n` to `type`. `type` feels more descriptive and since the `Person` type has a `name` property, `n` could be conflated to have something to do with that rather than the name of the type.